util/artifacts.go

```diff
@@ -14,82 +14,84 @@ import (
 type ArchiveContentFile struct {
 	Path string
 	Info os.FileInfo
+	err  error
 }
 
 func FindContentsToArchive(ctx context.Context, rootPath string, includes, excludes []string) ([]ArchiveContentFile, error) {
 	out := []ArchiveContentFile{}
+	catcher := grip.NewBasicCatcher()
+	for fn := range streamArchiveContents(ctx, rootPath, includes, excludes) {
+		if fn.err != nil {
+			catcher.Add(fn.err)
+			continue
+		}
 
-	files, errs := streamArchiveContents(ctx, rootPath, includes, excludes)
-	for fn := range files {
 		out = append(out, fn)
 	}
 
-	err, ok := <-errs
-	if ok && err != nil {
-		return nil, err
+	if catcher.HasErrors() {
+		return nil, catcher.Resolve()
 	}
 
 	return out, nil
 }
 
-func streamArchiveContents(ctx context.Context, rootPath string, includes, excludes []string) (<-chan ArchiveContentFile, <-chan error) {
-	errChan := make(chan error)
+func streamArchiveContents(ctx context.Context, rootPath string, includes, excludes []string) <-chan ArchiveContentFile {
 	outputChan := make(chan ArchiveContentFile)
 
 	go func() {
-		defer close(errChan)
 		defer close(outputChan)
 
 		for _, includePattern := range includes {
 			dir, filematch := filepath.Split(includePattern)
 			dir = filepath.Join(rootPath, dir)
 			exists, err := FileExists(dir)
 			if err != nil {
-				errChan <- err
+				outputChan <- ArchiveContentFile{err: err}
```

Comment:
continue

Reply:
did it, but I don't think it matters because exists is always false when there's an error, so I think it was fine. this was the existing logic.